lib/crawly/worker.ex

```diff
@@ -66,7 +70,15 @@ defmodule Crawly.Worker do
              response: HTTPoison.Response.t(),
              result: {:ok, response, spider_name} | {:error, term()}
   defp get_response({request, spider_name}) do
-    case HTTPoison.get(request.url, request.headers, request.options) do
+    # check if spider-level fetcher is set. Overrides the globally configured fetcher.
+    # if not set, log warning for explicit config preferred, get the globally-configured fetcher. Defaults to FwtchWithHTTPoison
+    fetcher = Application.get_env(
```

Comment:
Maybe we should allow tuple-style fetcher declarations too, to pass default options to the fetcher.

Reply:
+1